### ISSUE TITLE
Bug: no-unnecessary-condition: false positives for react-hook-form with zod integration for array path selector access

### ISSUE THREAD

**Issue by osdiab** [oxc-project/tsgolint#790]
Repro: install react-hook-form and the zod resolver.

```
import { useForm } from 'react-hook-form';
import { zodResolver } from '@hookform/resolvers/zod';
import { z } from 'zod'; // or 'zod/v4'

const schema = z.object({
  data: z.array(
    z
      .object({
        action: z.literal("create"),
        value: z.number(),
      })
      .nullish(),
  ),
});

function useMyForm() {
  // Automatically infers the output type from the schema
  const form = useForm({
    resolver: zodResolver(schema),
  });
  const theId = form.getValues(`data.0`)?.value;
}
```

`react-hook-form` path types are pretty intense so this might be doing something wild under the hood. but TypeScript recognizes that the form.getValues() call can return an undefined or null value.
